packages/opentelemetry-resources/src/platform/node/detect-resources.ts

```diff
@@ -34,6 +40,24 @@ export const detectResources = async (): Promise<Resource> => {
       }
     })
   );
+  if (config.logger) {
+    resources.forEach((resource, index) => {
```

Comment:
What do you think about extracting this logic into a `logResources` function and calling it here?

Reply:
Good idea, I've separated it out.